src/Core/src/Platform/Windows/ViewExtensions.cs

```diff
@@ -358,5 +358,29 @@ internal static Graphics.Rectangle GetBoundingBox(this FrameworkElement? nativeV
 
 			return null;
 		}
+
+		internal static IWindow? GetHostedWindow(this IView? view)
+			=> GetHostedWindow(view?.Handler?.NativeView as FrameworkElement);
+
+		internal static IWindow? GetHostedWindow(this FrameworkElement? view)
+			=> GetWindowForXamlRoot(view?.XamlRoot);
+
+		internal static IWindow? GetWindowForXamlRoot(XamlRoot? root)
+		{
+			if (root is null)
+				return null;
+
+			var windows = WindowExtensions.GetWindows();
+			foreach(var window in windows)
+			{
+				if (window.Handler?.NativeView is Microsoft.UI.Xaml.Window win)
+				{
+					if (win.Content.XamlRoot == root)
```

Comment:
Need to handle the case of a null content window.

Reply:
The analysers said that Content shouldn't be null, but it's also easy enough to add the check just to make sure, and since root already has a null check above it that should be okay. Fixed.